trieste/acquisition/rule.py

```diff
@@ -442,7 +443,9 @@ def __init__(
             )
 
         if builder is None:
-            builder = BatchMonteCarloExpectedImprovement(10_000)
+            builder = cast(
+                SingleModelAcquisitionBuilder[M_contra], BatchMonteCarloExpectedImprovement(10_000)
```

Comment:
Not sure why we need to tell mypy this (I though it should be able to infer it from the first overload).